### PR TITLE
Prepare 10.1 release

### PR DESCRIPTION
**Pull request by ryyppy**
- [x] Async / await docs
- [x] Update Promise docs
- [x] Blog post announcement
- [x] Update bsconfig.json schema file (fixed on master)
- [x] Update JS cheatsheet
- [x] Update syntax widget
- [x] ~~Playground bundle~~ (cherry-picked to main)

### CODE REVIEW

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  Async/await & better Promise support, JSX v4, and more!
+---
+
+## Introduction
+
+We are happy to announce ReScript 10.1!
+
+ReScript is a robustly typed language that compiles to efficient and human-readable JavaScript. It comes with one of the fastest build toolchains and offers first class support for interoperating with ReactJS and other existing JavaScript code.
+
+Use `npm` to install the newest [10.1 release](https://www.npmjs.com/package/bs-platform/v/10.1.1):
```
Comment (aspeddro) [inline _blogposts/2023-02-02-release-10-1.mdx:16]: rename bs-platform -> rescript

Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx]: fixed

File: pages/docs/manual/latest/async-await.mdx [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+  await GlobalAPI.sendAnalytics("User details have been logged for " + userId + "");
+  console.log("Email address for user " + userId + ": " + email + "");
+}
+```
+
+</CodeTab>
+
+As we can see above, an `async` function is defined via the `async` keyword right before the function's parameter list. In the function body, we are now able to use the `await` keyword to explicitly wait for a `Promise` value and assign its content to a let binding `email`.
+
+You will probably notice that this looks very similar to `async` / `await` in JS, but there's still a few details that are specific to ReScript. The next few sections will go through all the details that are specific to the ReScript type system.
```
Comment (cknitt) [inline pages/docs/manual/latest/async-await.mdx:60]:
```suggestion
You will probably notice that this looks very similar to `async` / `await` in JS, but there are still a few details that are specific to ReScript. The next few sections will go through all the details that are specific to the ReScript type system.
```

File: pages/docs/manual/latest/async-await.mdx [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+## Basics
+
+- You may only use `await` in `async` function bodies
+- `await` may only be called on a `promise` value
+- `await` calls are expressions, therefore they can be used in pattern matching (`switch`)
+- A function returning a `promise<'a>` is equivalent to an `async` function returning a value `'a` (important for writing signature files and bindings)
+- `promise` values and types returned from an `async` function don't auto-collapse
```
Comment (cknitt) [inline pages/docs/manual/latest/async-await.mdx:68]: What does that mean?

Comment (cknitt) [reply pages/docs/manual/latest/async-await.mdx:68]: Ah, it's explained further below.

File: pages/docs/manual/latest/async-await.mdx [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+```
+
+</CodeTab>
+
+## `await` multiple promises
+
+We can utilize the `Js.Promise2` module to handle multiple promises. E.g. let's use `Js.Promise2.all` to wait for multiple promises before continuing the program:
+
+```res
+let pauseReturn = async (value, timeout) => {
```
Comment (cknitt) [inline pages/docs/manual/latest/async-await.mdx:260]: This example is weird as `pauseReturn` returns a `promise<promise<'a>>`.

I think it should be

```rescript
let pauseReturn = (value, timeout) => {
  Js.Promise2.make((~resolve, ~reject) => {
    Js.Global.setTimeout(() => {
      resolve(. value)
    }, timeout)->ignore
  })
}

let logMultipleValues = async () => {
  let promise1 = pauseReturn("value1", 2000)
  let promise2 = pauseReturn("value2", 1200)
  let promise3 = pauseReturn("value3", 500)

  let all = await Js.Promise2.all([promise1, promise2, promise3])

  switch all {
  | [v1, v2, v3] => Js.log(`All values: ${v1}, ${v2}, ${v3}`)
  | _ => Js.log("this should never happen")
  }
}
```

**Review by cknitt — CHANGES_REQUESTED** [review]

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+```resi
+// User.resi
+type user
+
+let fetchUser: string => promise<user>
+```
+
+Way easier on the eyes, don't you think? Note that the new `promise` type is fully compatible with `Js.Promise.t` (no breaking changes).
+
+Additionally, we also introduced the `Js.Promise2` module as a stepping stone to migrate `Js.Promise` based code to a first-pipe (->) friendly solution.
```
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:71]: Yes we could do that. Will think about it.

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+(*Sidenote*: We know that our users want a solution to unify `Belt`, `Js` and `Js.xxx2` and have a fully featured "standard library". Good news is that we have a solution in the pipeline to fix this. `Js.Promise2` was introduced to ease the process later on and is not supposed to be the panacea of promise handling.)
+
+If you are already using a third-party promise library like [ryyppy/rescript-promise](https://github.com/ryyppy/rescript-promise) or similar, there's no need to migrate, but we highly encourage you to mix and match it with `async` / `await`.
+
+
+## New JSX v4 syntax
+
+ReScript 10.1 now ships with JSX v4. Here's what's new:
```
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:80]: Good point. Will add.

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+  // "classic" mode will compile jsx to React.createElement calls
+  @@jsxConfig({version: 4, mode: "classic"})
+
+  @react.component
+  let make = (~name) => {
+    <div> {React.string(`Hello ${name}`)} </div>
+  }
+}
+
+module NoDecoratorExample = {
```
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:117]: We can rename it.

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+
+### How to migrate to JSX v4?
+
+We provide a full [migration guide](/docs/react/latest/migrate-react) with all the details of an migration.
+
+Make sure to also check out the [rescript-react changelog](https://github.com/rescript-lang/rescript-react/blob/master/CHANGELOG.md) as well.
+
+## What's next?
+
+Our contributors are already one step ahead and work on major improvements in our next upcoming major release (v11). Things that are currently being explored:
```
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:287]: Eh, sounds weird.. maybe...

```suggestion
Our contributors are already one step ahead and are currently working on improvements for the next major v11 release. Things that are currently being explored:
```

File: pages/docs/manual/latest/async-await.mdx (hunk 3 above)
Comment (ryyppy) [reply pages/docs/manual/latest/async-await.mdx:68]: Let's add this maybe...
```suggestion
- `promise` values and types returned from an `async` function don't auto-collapse (more on this later)
```

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+  @@jsxConfig({version: 4, mode: "automatic"})
+
+  @react.component
+  let make = () => {
+    let props = {NoDecoratorExample.name: "World"}
+
+    <NoDecoratorExample {...props} />
+  }
+}
+
```
Comment (ryyppy) [inline _blogposts/2023-02-02-release-10-1.mdx:169]: Add another example for better type inference on nominal types:
https://rescript-lang.org/try?version=v10.1.2&code=ALBWGcA8GEHsDsBmBLA5gCgN4AIBuBTAJ3GQQC5sBmAGmwFtYATfCgIgEMBXAF1jve7IAxq2wBfAJQAoKQ0acANvmwBVcEWwBebJinZs3AJ4AHZeG4DO4LdgDy8BcnjKAPncSJHzvdh-BC+OxC3AB0QnzGCPjw3D5K3PTsANbK2ugAfuaW4BJaAHw6PvrgAO7I3EIAFthZ3FaF+o3YbvZeqQUASoHBIfCKCkWNLR5t+dhdQaF9CgNNYj7z81Lx2AD6NgA8ahq1VpqtTsoA9HlSQA

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+## New JSX v4 syntax
+
+ReScript 10.1 now ships with JSX v4. Here's what's new:
+
+- **Cleaner interop.** Due to recent improvements in the type checker, the `@react.component` transformation doesn't require any `makeProps` convention anymore. `make` functions will now be transformed into a `prop` type and a component function. That's it.
+- **Two new transformation modes**. JSX v4 comes with a `classic` mode (= `React.createElement`) and `automatic` mode (= `jsx-runtime` calls). The latter is the new default, moving forward with `rescript/react@0.11` and `React@18`.
+- **Allow mixing JSX configurations on the project and module level.** Gradually mix and match JSX transformations and modes without migrating any old code!
+- **Pass `prop` types** to `@react.component`. You can now fine tune `@react.component` with your specific prop type needs. Very useful for libraries and frameworks to define component interfaces.
+- **Revisited props spread operator.** This will allow users to spread records in JSX without sacrificing their sanity. Note that this implementation has harder constraints than its JS counterpart. (requires `rescript/react@0.11` or higher)
+
```
Comment (zth) [inline _blogposts/2023-02-02-release-10-1.mdx:87]:
```suggestion
- **Paving the way for better editor tooling.** The new representation in JSX v4 opens up for better editor tooling, like providing typed completion for values and callbacks of JSX props.
- **Better type inference of props.** Type inference when passing for example variants that are defined in the same module as the component is much improved. With the earlier JSX version, you'd often need to write code like this in order for the compiler to understand which variant you're passing: `<Button variant=Button.Primary text="Click" />`. With JSX v4, you won't need to tell the compiler where the variant you're passing is located: `<Button variant=Primary text="Click" />`

```

**Review by zth — COMMENTED** [review]
Just 2 suggestions, do with them what you want.

File: _blogposts/2023-02-02-release-10-1.mdx [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+}
+```
+</CodeTab>
+
+### How to migrate to JSX v4?
+
+We provide a full [migration guide](/docs/react/latest/migrate-react) with all the details of an migration.
+
+Make sure to also check out the [rescript-react changelog](https://github.com/rescript-lang/rescript-react/blob/master/CHANGELOG.md) as well.
+
```
Comment (ryyppy) [inline _blogposts/2023-02-02-release-10-1.mdx:284]: Add acknowledgements for the JSX@v4 work

File: _blogposts/2023-02-02-release-10-1.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx]: done

File: _blogposts/2023-02-02-release-10-1.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:80]: done.

File: pages/docs/manual/latest/async-await.mdx (hunk 4 above)
Comment (ryyppy) [reply pages/docs/manual/latest/async-await.mdx]: done.

File: _blogposts/2023-02-02-release-10-1.mdx (hunk 9 above)
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:172]: no need anymore, the improvement has been mentioned above already.

File: _blogposts/2023-02-02-release-10-1.mdx (hunk 11 above)
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx:287]: done.

File: _blogposts/2023-02-02-release-10-1.mdx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ryyppy) [reply _blogposts/2023-02-02-release-10-1.mdx]: done.

**Pull request merged**